src/time.rs

```diff
@@ -79,11 +80,13 @@ pub struct MonoTimer {
 
 impl MonoTimer {
     /// Creates a new `Monotonic` timer
-    pub fn new(mut dwt: DWT, clocks: Clocks) -> Self {
+    pub fn new(mut dwt: DWT, mut dcb: DCB, clocks: Clocks) -> Self {
+        dcb.enable_trace();
         dwt.enable_cycle_counter();
 
         // now the CYCCNT counter can't be stopped or resetted
         drop(dwt);
+        drop(dcb);
```

Comment:
Wouldn't it be better to have it in Clocks ? Like that you can use it in other things as in the I2C driver ?

Reply:
Hi, sorry but I dont unerstand what you mean. Could you give a more concrete example?

Reply:
I suppose other things than MonoTimer can rely on trace being enable. If the creation of MonoTimer eat DCB, no other code can check that trace is enable. Thus maybe we can have the info somewhere. I ws thinking in the Clock object, but that may not be appropriate.

Reply:
An idea would be to have something like the `Clocks` object: a function that takes DCB, enable trace and return a copyable token that guaranties that trace is enable.

```rust
#[derive(Copy, Clone)]
pub struct TraceEnable {
    _0: (),
}
pub fn enable_trace(mut dcb: DCB) -> TraceEnable {
    dcb.enable_trace();
    TraceEnable { _0: () }
}
```

Then, your `MonoTimer` can take a `TraceEnable` parameter, as anything relying on trace to be enable.

Reply:
I'll look into it again, hopefully on sunday. It seems to be reasonable to do it the way you suggested it :)